src/vs/workbench/contrib/debug/browser/debugHover.ts

```diff
@@ -176,7 +182,52 @@ export class DebugHoverWidget implements IContentWidget {
 			options: DebugHoverWidget._HOVER_HIGHLIGHT_DECORATION_OPTIONS
 		}]);
 
-		return this.doShow(pos, expression, focus);
+		return expression;
+	}
+
+	private async showSelectionEvaluation(selection: Selection, model: ITextModel, session: IDebugSession): Promise<IExpression | undefined> {
+		if (!session.capabilities.supportsEvaluateForHovers) {
```

Comment:
You do not have to check this capability, it is only used by debuggers who want every hover to be an evaluate (selection not needed in that case).